src/compile/legend/rules.ts

```diff
@@ -30,8 +31,7 @@ export function type(legend: Legend, fieldDef: FieldDef, channel: Channel) {
   if (legend.type) {
     return legend.type;
   }
-
-  if (channel === COLOR && !fieldDef.bin && !fieldDef.timeUnit && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
+  if (channel === COLOR && !fieldDef.bin && (!fieldDef.timeUnit || containsTimeUnit(fieldDef.timeUnit, 'year')) && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
```

Comment:
Should we use scale type here (scale.type === `time` or `utc`)?  (Instead of checking for time prefix directly)